GooglePlayInstant/Editor/AndroidManifest/AndroidManifestHelper.cs

```diff
@@ -0,0 +1,264 @@
+            if (androidAttribute == null)
+            {
+                return PreconditionMissingXmlnsAndroid;
+            }
+
+            if (androidAttribute.Value != AndroidNamespaceUrl)
+            {
+                return PreconditionInvalidXmlnsAndroid;
+            }
+
+            // TSV2 is required for instant apps starting with Android Oreo.
+            manifestElement.SetAttributeValue(AndroidTargetSandboxVersionXName, "2");
+
+            return uri == null ? null : AddDefaultUrl(manifestElement, uri);
+        }
+
+        private static string AddDefaultUrl(XElement manifestElement, Uri uri)
+        {
+            var applicationElement = GetExactlyOne(manifestElement.Elements(Application));
+            if (applicationElement == null)
+            {
+                return PreconditionOneApplicationElement;
+            }
+
+            var mainActivity = GetExactlyOne(GetMainActivities(applicationElement));
+            if (mainActivity == null)
+            {
+                return PreconditionOneMainActivity;
+            }
+
+            var updateViewIntentFilterResult = UpdateViewIntentFilter(mainActivity, uri);
+            if (updateViewIntentFilterResult != null)
+            {
+                return updateViewIntentFilterResult;
+            }
+
+            return UpdateDefaultUrlElement(mainActivity, uri);
+        }
+
+        private static string UpdateViewIntentFilter(XElement mainActivity, Uri uri)
+        {
+            // Find the Activity's Intent Filter with action type VIEW to update, or create a new Intent Filter.
+            var actionViewIntentFilters = GetActionViewIntentFilters(mainActivity);
+            XElement viewIntentFilter;
+            switch (actionViewIntentFilters.Count())
+            {
+                case 0:
+                    viewIntentFilter = new XElement(IntentFilter);
+                    mainActivity.Add(viewIntentFilter);
+                    break;
+                case 1:
+                    viewIntentFilter = actionViewIntentFilters.First();
+                    // TODO: preserve existing elements and just update
+                    viewIntentFilter.RemoveAll();
+                    break;
+                default:
+                    // TODO: add support for activities with multiple VIEW intent filters
+                    return "more than one VIEW intent-filter";
+            }
+
+            // See https://developer.android.com/topic/google-play-instant/getting-started/game-instant-app#app-links
+            // and https://developer.android.com/training/app-links/verify-site-associations for info on "autoVerify".
+            viewIntentFilter.SetAttributeValue(AndroidAutoVerifyXName, "true");
+            viewIntentFilter.Add(CreateElementWithAttribute(Action, AndroidNameXName, ActionView));
+            viewIntentFilter.Add(
+                CreateElementWithAttribute(Category, AndroidNameXName, "android.intent.category.BROWSABLE"));
```

Comment:
nit: we should make android.intent.category.BROWSABLE and DEFAULT below into constants like we did for the VIEW action.

Reply:
Done. (Just to explain why it was this way, I created constants for strings used 2+ times and not for strings used once.)